signer/src/utxo.rs

```diff
@@ -0,0 +1,964 @@
+use bitcoin::absolute::LockTime;
+use bitcoin::key::Secp256k1;
+use bitcoin::secp256k1::SECP256K1;
+use bitcoin::taproot::LeafVersion;
+use bitcoin::taproot::NodeInfo;
+use bitcoin::taproot::Signature;
+use bitcoin::taproot::TaprootSpendInfo;
+use bitcoin::transaction::Version;
+use bitcoin::Amount;
+use bitcoin::OutPoint;
+use bitcoin::PublicKey;
+use bitcoin::ScriptBuf;
+use bitcoin::Sequence;
+use bitcoin::Transaction;
+use bitcoin::TxIn;
+use bitcoin::TxOut;
+use bitcoin::Witness;
+use bitcoin::XOnlyPublicKey;
+use secp256k1::Keypair;
+use secp256k1::Message;
+
+use crate::error::Error;
+use crate::packaging::compute_optimal_packages;
+use crate::packaging::Weighted;
+
+#[derive(Debug, Clone, Copy)]
+pub struct SignerBtcState {
+    /// The outstanding signer UTXO.
+    pub utxo: SignerUtxo,
+    /// The current market fee rate in sat/vByte.
+    pub fee_rate: u64,
+    /// The current public key of the signers
+    pub public_key: PublicKey,
+}
+
+#[derive(Debug)]
+pub struct SbtcRequests {
+    /// Accepted and pending deposit requests.
+    pub deposits: Vec<DepositRequest>,
+    /// Accepted and pending withdrawal requests.
+    pub withdrawals: Vec<WithdrawalRequest>,
+    /// Summary of the Signers' UTXO and information necessary for
+    /// constructing their next UTXO.
+    pub signer_state: SignerBtcState,
+    /// The maximum acceptable number of votes against for any given
+    /// request.
+    pub reject_capacity: u32,
```

Comment:
Yeah, it is more intuitive. I think I had that design originally but with an additional field noting the total number of signers so that we could compute the reject capacity (the accept threshold alone is not enough). I'm okay with switching that version of things with the additional field.